### PR TITLE
All groups for output

### PR DESCRIPTION
**Pull request by jnthn**
This hopefully provides a solution to #299. If you call `.results-by-filter-group(:all)` (note the new ':all` named argument), then it will include all possible filter keys that could exist (based on the `enum`s in the definitions) rather than just those that have been used in an instance. A `0` value is given to those that have no instances.

This is achieved by dynamic means: filter group collections are constructed with the filter set(s) from the model of all modules that contribute values to the group. This is mostly a case of passing it along, with the exception of pairwise operations which need to perform a union of the module filter sets (thus how a `P+` involving two different modules will end up with the filter sets of both being accounted for).

Hopefully this provides the desired semantics, and also will make it an easy change in the output productions too (I did not update those; @zaucker it's probably best to check out this branch and have a go at updating the output logic and seeing if it does the right thing).

### CODE REVIEW

**Comment by jnthn** [comment]
This also probably helps with #301 in so far as the code in that area is cleaned up by this PR (although it still needs attention).

**Comment by zaucker** [comment]
Will extend the other OutputFormatters in separate PR.

File: lib/Agrammon/Outputs/FilterGroupCollection.pm6 [hunk 1]
```diff
@@ -54,30 +57,57 @@ class Agrammon::Outputs::FilterGroupCollection {
...
+    #| group. If :all is passed, then all filters that could possibly have been
+    #| selected will be included, even if no instance used them, and they will
+    #| have a zero value.
+    method results-by-filter-group(Bool :$all = False) {
+        if $all {
+            if %!values-by-filter && !$!provenance {
+                die "Can only get all values when provenance of the filter group was provided";
+            }
+            my @results;
+            for $!provenance.keys.sort.reverse -> $filter-set {
```
Comment (jnthn) [inline lib/Agrammon/Outputs/FilterGroupCollection.pm6:70]: Unfortunately, this `sort` is not going to do anything useful. It is a `sort` on instances of `Agrammon::Model::FilterSet`, and will sort on memory location or some such. :-)

Comment (jnthn) [reply lib/Agrammon/Outputs/FilterGroupCollection.pm6]: I did 437fbcc18 but I don't know if it's what you actually want. :-)

File: lib/Agrammon/Outputs/FilterGroupCollection.pm6 [hunk 2]
```diff
@@ -54,30 +57,57 @@ class Agrammon::Outputs::FilterGroupCollection {
...
+    #| group. If :all is passed, then all filters that could possibly have been
+    #| selected will be included, even if no instance used them, and they will
+    #| have a zero value.
+    method results-by-filter-group(Bool :$all = False) {
+        if $all {
+            if %!values-by-filter && !$!provenance {
+                die "Can only get all values when provenance of the filter group was provided";
+            }
+            my @results;
+            for $!provenance.keys.sort(*.module.taxonomy).reverse -> $filter-set {
```
Comment (zaucker) [inline lib/Agrammon/Outputs/FilterGroupCollection.pm6:70]: This sort does almost what we need. Will make a separate PR for it,

**Review by zaucker — APPROVED** [review]
